app/src/main/java/org/simple/clinic/scanid/ui/ShortCodeSpanWatcher.kt

```diff
@@ -0,0 +1,56 @@
+package org.simple.clinic.scanid.ui
+
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.text.Editable
+import android.text.Spannable
+import android.text.style.ReplacementSpan
+import kotlin.math.roundToInt
+
+private const val CHARS_REQUIRED_TO_INSERT_SPACE = 4
+private const val INSERT_SPACE_AT_INDEX = 2
+
+class ShortCodeSpanWatcher : TextWatcherAdapter() {
+    private var activeSpan: AddTrailingSpaceSpan? = null
+
+    override fun afterTextChanged(editable: Editable) {
+        with(editable) {
+            if (length >= CHARS_REQUIRED_TO_INSERT_SPACE) {
+                removeSpan(activeSpan)
+                activeSpan = AddTrailingSpaceSpan()
+                setSpan(activeSpan, INSERT_SPACE_AT_INDEX, INSERT_SPACE_AT_INDEX + 1, Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
+            } else {
+                removeSpan(activeSpan)
+                activeSpan = null
+            }
+        }
+    }
+}
+
+private class AddTrailingSpaceSpan : ReplacementSpan() {
```

Comment:
This is something that can be used even when formatting the title for the Scan BP passport screens. Can we break this out into a separate class? We already have a package where we keep our custom spans. (`org.simple.clinic.text.style`).